src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests5.cs

```diff
@@ -680,59 +1346,401 @@ class C {
+    {
+        if (c1 is { Prop1: null } &&
+            c1 is { Prop1.Prop2: null })
+        {
+            c1.Prop1.ToString();
+            c1.Prop1.Prop2.ToString(); // 2
+        }
+        else
+        {
+            c1.Prop1.ToString(); // 3
+            c1.Prop1.Prop2.ToString();
+        }
+    }
+
+    public void M4(C1 c1)
+    {
+        if (c1 is { Prop1: null } ||
+            c1 is { Prop1.Prop2: null })
+        {
+            c1.Prop1.ToString();      // 4
+            c1.Prop1.Prop2.ToString(); // 5
+        }
+        else
+        {
+            c1.Prop1.ToString();
+            c1.Prop1.Prop2.ToString();
+        }
+    }
+}
+
+class C1 { public C2 Prop1 = null!; }
+class C2 { public object Prop2 = null!; }
+";
+            var compilation = CreateCompilation(program, parseOptions: TestOptions.RegularWithExtendedPropertyPatterns);
+            compilation.VerifyEmitDiagnostics(
+                // (10,13): warning CS8602: Dereference of a possibly null reference.
+                //             c1.Prop1.Prop2.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c1.Prop1.Prop2").WithLocation(10, 13),
+                // (34,13): warning CS8602: Dereference of a possibly null reference.
+                //             c1.Prop1.Prop2.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c1.Prop1.Prop2").WithLocation(34, 13),
+                // (38,13): warning CS8602: Dereference of a possibly null reference.
+                //             c1.Prop1.ToString(); // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c1.Prop1").WithLocation(38, 13),
+                // (48,13): warning CS8602: Dereference of a possibly null reference.
+                //             c1.Prop1.ToString();      // 4
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c1.Prop1").WithLocation(48, 13),
+                // (49,13): warning CS8602: Dereference of a possibly null reference.
+                //             c1.Prop1.Prop2.ToString(); // 5
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c1.Prop1.Prop2").WithLocation(49, 13)
+                );
+        }
+
+        [Fact]
+        public void ExtendedPropertyPatterns_ExpressionColonInPositionalPattern()
+        {
+            var source = @"
+class C
+{
+    C Property { get; set; }
+
+    void M()
+    {
+        _ = this is (Property.Property: null, Property: null);
+    }
+
+    public void Deconstruct(out C c1, out C c2)
+        => throw null;
+}
+";
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.Regular9);
+            compilation.VerifyEmitDiagnostics(
+                // (8,22): error CS8652: The feature 'extended property patterns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         _ = this is (Property.Property: null, Property: null);
```

Comment:
>Property.Property [](http://example.com/codeflow?start=40&length=17)

It seems surprising that we're parsing this as an extended property pattern, rather than just a syntax error at the `'.'`. Is there a reason to allow nested properties here?

Reply:
Both positional and recurisve patterns use the same syntax node `SubpatternSyntax` and share the same parsing logic (`ParseSubpatternElement`).
There's no reason beyond that.